### ISSUE TITLE
`docker-compose up -d` fails to build because of download error

### ISSUE THREAD

**Issue by pangard** [naturalcrit/homebrewery#3111]
### Renderer

N/A

### Browser

N/A

### Operating System

Linux

### What happened?

Following instructions from `README.DOCKER.md`, `docker-compose up -d` fails like this

```
> docker-compose down -v
Removing network homebrewery_default
Removing volume homebrewery_mongodata
> docker-compose up -d
Creating network "homebrewery_default" with the default driver
Creating volume "homebrewery_mongodata" with default driver
Building homebrewery
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
            Install the buildx component to build images with BuildKit:
            https://docs.docker.com/go/buildx/

Sending build context to Docker daemon  21.82MB
Step 1/10 : FROM node:18-alpine
 ---> d1517ab6615b
Step 2/10 : RUN apk --no-cache add git
 ---> Running in c6ca5454905c
fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/community/x86_64/APKINDEX.tar.gz
WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.18/main: temporary error (try again later)
WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.18/community: temporary error (try again later)
ERROR: unable to select packages:
  git (no such package):
    required by: world[git]
The command '/bin/sh -c apk --no-cache add git' returned a non-zero code: 1
ERROR: Service 'homebrewery' failed to build : Build failed
```
## Context
```bash
> docker-compose version
docker-compose version 1.29.2, build unknown
docker-py version: 5.0.3
CPython version: 3.10.12
OpenSSL version: OpenSSL 3.0.2 15 Mar 2022
```
 - OS: `Ubuntu 22.04.3 LTS`
 - `linux kernel: 5.15.0-86-generic #96-Ubuntu SMP`
 - docker: `Docker version 24.0.5, build 24.0.5-0ubuntu1~22.04.1`

Regards!

**Comment by pangard** [comment]
I reinstalled docker in a different way and updated to 

`FROM node:21-alpine`

and it builds fine. Then docker seems to be running, but I can't find the server. Both `sudo netstat -tulpn` and `sudo systemctl status docker` hint at a few locations such as 127.0.0.1:8000, but nothing shows up in the browser `:-/`.

Then `docker logs homebrewery_homebrewery_1`

shows

```
ReferenceError: document is not defined
    at /usr/src/app/node_modules/codemirror/lib/codemirror.js:88:3
    at /usr/src/app/node_modules/codemirror/lib/codemirror.js:11:83
    at Object.<anonymous> (/usr/src/app/node_modules/codemirror/lib/codemirror.js:14:2)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at o (/usr/src/app/build/homebrew/ssr.js:1:499)
```

Anyway, I'll close this issue and maybe later open a new one, when I can research this. Sorry for the noise

**Comment by afbagwell** [comment]
> ```
> ReferenceError: document is not defined
>     at /usr/src/app/node_modules/codemirror/lib/codemirror.js:88:3
>     at /usr/src/app/node_modules/codemirror/lib/codemirror.js:11:83
>     at Object.<anonymous> (/usr/src/app/node_modules/codemirror/lib/codemirror.js:14:2)
>     at Module._compile (node:internal/modules/cjs/loader:1376:14)
>     at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
>     at Module.load (node:internal/modules/cjs/loader:1207:32)
>     at Module._load (node:internal/modules/cjs/loader:1023:12)
>     at Module.require (node:internal/modules/cjs/loader:1235:19)
>     at require (node:internal/modules/helpers:176:18)
>     at o (/usr/src/app/build/homebrew/ssr.js:1:499)
> ```
> 
> Anyway, I'll close this issue and maybe later open a new one, when I can research this. Sorry for the noise

I'm not using Docker, but I just ran into this exact error and stacktrace myself trying to run homebrewery on WSL/Ubuntu. Yours was the only issue with a text match.

Running with node 21.1.0 and mongodb 7.0.2

**Comment by pangard** [comment]
Afraid to say I don't really understand what happened `:-( `
BUT I got it to work by:
1.  deleting everything docker related (using `docker image ls`, `docker container ls`, `docker container stop CONTAINER_ID`, `docker container rm CONTAINER_ID`, `docker image rm IMAGE_ID`)
2. uninstalling and purging docker
3. installing it clean from scratch as in this [tutorial for ubuntu 22.04](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-on-ubuntu-22-04)
4. git clone this repo
5. `docker compose up -d`

I think one source of trouble is that since the tutorial mentioned "docker-compose", I installed "docker-compose", while standard modern docker is enough, just doing `docker compose up -d`  instead of `docker-compose up -d`.
